### ISSUE TITLE
Orders are not getting aknowledged

### ISSUE THREAD

**Issue by eddiedale** [elleracompany/commerce-klarna-checkout#70]
Currently solving this in a custom module as follows:

```
use craft\commerce\Plugin as Commerce;

Event::on(Order::class, Order::EVENT_AFTER_COMPLETE_ORDER, function (Event $e) {

  $order = $e->sender;

  $hash = Craft::$app->request->getQueryParam('hash');
  
  $plugin = Commerce::getInstance();
  $last_transaction = $plugin->getTransactions()->getTransactionByHash($hash);
  
  $gateway = $plugin->getGateways()->getGatewayById($last_transaction->gatewayId);
  $klarna_order_id = $last_transaction->reference;
  
  // Acknowledge the order
  $acknowledgement = new Acknowledge($gateway, $klarna_order_id);

}
```

**Comment by jellingsen** [comment]
Is this with the gateway in "purchase" mode?

Seems like we're only sending the Acknowledge request on gateways in "authorize" mode.

**Comment by aelvan** [comment]
@eddiedale Think maybe I'm seeing the same thing, how does this manifest for you in commerce/klarna? 

The issue I'm having is that occasionally (five orders over the last 14 days) I'm getting an order in commerce that is completed, but in the list of transactions I only have the "Authorize - pending" entry, not the "Authorize - success" one. So the order can't be captured, but the customer has gotten a confirmation about the order being completed. Does it sound like the same issue?

**Comment by eddiedale** [comment]
@aelvan No, I think that is a different issue. At least I do not have problems with any orders/authorizations pending.

@jellingsen Yes, the gateway is in "purchase" mode, and indeed the acknowledgment works as it should if I change it to "Only authorize", and manually capture thereafter. I do want to keep it in purchase mode though, so it would be great with an update solving that.
